cmd/agent/app/integrations.go

```diff
@@ -390,11 +391,36 @@ func downloadWheel(integration, version string) (string, error) {
 	if verbose > 0 {
 		args = append(args, fmt.Sprintf("-%s", strings.Repeat("v", verbose)))
 	}
+
 	downloaderCmd := exec.Command(pyPath, args...)
+
 	// Change the working directory to one the Datadog Agent can read, so that we
 	// can switch to temporary working directories, and back, for in-toto.
 	downloaderCmd.Dir, _ = executable.Folder()
-	downloaderCmd.Env = os.Environ()
+
+	// We do all of the following so that when we call our downloader, which will
+	// in turn call in-toto, which will in turn call Python to inspect the wheel,
+	// we will use our embedded Python.
+	// First, get the current PATH as an array.
+	path_arr := filepath.SplitList(os.Getenv("PATH"))
+	// Get the directory of our embedded Python.
+	path_dir := filepath.Dir(pyPath)
+	// Prepend this dir to PATH array.
+	path_arr = append([]string{path_dir}, path_arr...)
+	// Build a new PATH string from the array.
+	path_str := strings.Join(path_arr, string(os.PathListSeparator))
+	// Make a copy of the current environment.
+	environ := os.Environ()
+	// Walk over the copy of the environment, and replace PATH.
+	for key, value := range environ {
+		if strings.HasPrefix(value, "PATH=") {
+			environ[key] = "PATH=" + path_str
+			break
+		}
+	}
+	// Now, while downloaderCmd itself won't use the new PATH, any child process,
```

Comment:
Not sure I understand why. You are setting `downloaderCmd` environment so it should have the new PATH.

Reply:
Yes, that is the expectation, but this is the counterintuitive behaviour :)